instant-xml/tests/all.rs

```diff
@@ -41,52 +35,104 @@ fn struct_with_named_fields() {
         }
         .to_xml()
         .unwrap(),
-        "<StructWithNamedFields xmlns=\"URI\" xmlns:bar=\"BAZ\" xmlns:foo=\"BAR\"><flag>true</flag><bar:string>test</bar:string><number xmlns=\"typo\">1</number></StructWithNamedFields>"
+        "<StructWithNamedFields xmlns=\"\" xmlns:bar=\"BAZ\" xmlns:foo=\"BAR\"><flag>true</flag><bar:string>test</bar:string><number xmlns=\"typo\">1</number></StructWithNamedFields>"
     );
 }
 
+#[derive(Debug, Eq, PartialEq, ToXml)]
+#[xml(namespace("URI", dar = "BAZ", internal = "INTERNAL"))]
+struct Nested {
+    #[xml(namespace(dar))]
+    flag_parent_prefix: bool,
+    #[xml(namespace(internal))]
+    flag_internal_prefix: bool,
+}
+
 #[derive(Debug, Eq, PartialEq, ToXml)]
 #[xml(namespace("URI", bar = "BAZ", foo = "BAR"))]
 struct StructWithCustomField {
+    #[xml(attribute)]
+    int_attribute: i32,
+    #[xml(namespace("BAZ"))]
+    flag_direct_namespace_same_as_prefix: bool,
```

Comment:
For this case, I think we should serialize ` xmlns="BAZ"` instead of using the prefix.

We're slightly overloading the meaning of the unnamed first namespace argument to mean both the namespace that an element is in *and* the default namespace for that element. These aren't quite the same thing, and what you have currently is correct, but I think it would be more consistent for now to use `BAR` as the default in this case.